models/attribution.js

```diff
@@ -0,0 +1,286 @@
+/*
+ * Most of this file was copied as-is (keeping function names intact)
+ * and refactored only slightly to make comparisons between this app and the
+ * original Lizenhinweisgenerator easier.
+ */
+
+const assert = require('assert');
+const jsdom = require('jsdom');
+const License = require('./license');
+const HtmlSaniziter = require('../services/htmlSanitizer');
+
+const { JSDOM } = jsdom;
+
+const KNOWN_LANGUAGES = ['en', 'es', 'pt', 'de', 'uk'];
+const KNOWN_TYPES_OF_USE = ['online', 'offline'];
+const ATTRIBUTION_TRANSLATIONS = {
+  en: {
+    'pd-attribution-hint': 'marked as public domain',
+    'check-details': 'more details on',
+    anonymous: 'anonymous',
+    by: 'by',
+    edited: 'modified',
+  },
+  es: {
+    'pd-attribution-hint': 'marcado como dominio público',
+    'check-details': 'para más detalles véase',
+    anonymous: 'anónimo',
+    by: 'por',
+    edited: 'modificado',
+  },
+  pt: {
+    'pd-attribution-hint': 'marcado como domínio público',
+    'check-details': 'para mais detalhes, veja',
+    anonymous: 'anónimo',
+    by: 'por',
+    edited: 'modificado',
+  },
+  de: {
+    'pd-attribution-hint': 'als gemeinfrei gekennzeichnet',
+    'check-details': 'Details auf',
+    anonymous: 'anonym',
+    by: 'von',
+    edited: 'bearbeitet',
+  },
+  uk: {
+    'pd-attribution-hint': 'позначено як суспільне надбання',
+    'check-details': 'більше деталей на',
+    anonymous: 'анонім',
+    by: 'автор',
+    edited: 'модифіковано',
+  },
+};
+
+function validationError(attribute) {
+  return `Attribution: Invalid "${attribute}" provided`;
+}
+
+function t(lang, key) {
+  return ATTRIBUTION_TRANSLATIONS[lang][key];
+}
+function isStringPresent(string) {
+  return typeof string === 'string' && string.length > 0;
+}
+
+function validateParams({
```

Comment:
🤔 I feel like we should introduce a wrapper object for `fileData`... Here you're expecting `fileUrl`, and `fileTitle` which are referred to as `rawUrl` and `title` in other services...

If we'd have a common wrapper we could simply assert that the thing we're receiving has this interface...

Background: figuring out how to rename the attributes when passing them in from other services is what's currently the most work for me when taking over #15 ...

Reply:
✅  i agree, implemented that change and commited it to this PR